tests/functional/targets/test_remove_targets.py

```diff
@@ -0,0 +1,200 @@
+    )
+    assert response.status_code == 202, response.text
+    threshold = 90
+    task_completed_within_threshold(
+        http_request, headers, response.json(), threshold
+    )
+    # Make sure all of the targets can be downloaded
+    for path in paths:
+        target_info = get_target_info(path)
+        assert target_info is not None
+
+
+@when(
+    parse("the API requester deletes all of the following targets {paths}"),
+    target_fixture="response",
+)
+def the_api_requester_deletes_targets(http_request, headers, paths):
+    # remove quotes; example "[str, str]" -> [str, str]
+    paths = ast.literal_eval(paths)
+    payload = {"targets": paths}
+    return http_request(
+        method="DELETE", url="/api/v1/targets", headers=headers, json=payload
+    )
+
+
+@then(
+    "the API requester should get status code '202' with 'task_id'",
+    target_fixture="response_json",
+)
+def the_api_requester_gets_successful_message(response):
+    assert response.status_code == 202, response.text
+    return response.json()
+
+
+@then(
+    "the API requester gets from endpoint 'GET /api/v1/task' status "
+    "'Task finished' within 90 seconds"
+)
+def the_api_requester_gets_task_status_finished_within_threshold(
+    http_request, headers, task_completed_within_threshold, response_json
+):
+    threshold = 90
+    task_completed_within_threshold(
+        http_request, headers, response_json, threshold
+    )
+
+
+@then(
+    parse(
+        "all of the targets {paths} should not be available for download "
+        "using TUF client from the metadata repository"
+    )
+)
+def the_targets_are_no_longer_available_for_download(get_target_info, paths):
+    # remove quotes; example "[str, str]" -> [str, str]
+    paths = ast.literal_eval(paths)
+    for path in paths:
+        target_info = get_target_info(path)
+        assert target_info is None
+
+
+@scenario(
+    "../../features/targets/remove_targets.feature",
+    "Removing targets that does exist and ignoring the rest",
+)
+def test_removing_targets_that_does_exist_and_ignoring_the_rest():
+    """Removing targets that does exist and ignoring the rest."""
+
+
+@when(
+    parse(
+        "the API requester tries to delete all of the following targets "
+        "{paths} and {non_existing_paths}"
+    ),
+    target_fixture="response",
+)
+def the_api_requester_tries_to_delete_all_targets(
+    http_request, headers, paths, non_existing_paths
+):
+    # remove quotes; example "[str, str]" -> [str, str]
+    paths = ast.literal_eval(paths)
+    non_existing_paths = ast.literal_eval(non_existing_paths)
+    payload = {"targets": paths + non_existing_paths}
+    return http_request(
+        method="DELETE", url="/api/v1/targets", headers=headers, json=payload
+    )
+
+
+@then(
+    "the API requester gets from endpoint 'GET /api/v1/task' status "
+    "'Task finished' within 90 seconds",
+    target_fixture="targets_dict",
+)
```

Comment:
This one can be reused from above (line 130), as you want to check only if the task finishes before the threshold.
The part  from lines 170 to 177, you can move to `the_api_requester_gets_deleted_and_not_found_lists()`